test/e2e/upgrades/apps/deployments.go

```diff
@@ -84,9 +85,6 @@ func (t *DeploymentUpgradeTest) Setup(f *framework.Framework) {
 
 	rs, err := deploymentutil.GetNewReplicaSet(deployment, c.ExtensionsV1beta1())
 	framework.ExpectNoError(err)
-	if rs == nil {
-		framework.ExpectNoError(fmt.Errorf("expected a new replica set for deployment %q, found none", deployment.Name))
-	}
 
 	// Store the old replica set - should be the same after the upgrade.
 	t.oldRS = rs
```

Comment:
Do we still need oldRS and newRS?

Reply:
In fact, we do not need them. Instead, i will create a new variable to record UID of old RS rather than storing the whole RS itself. The old RS is only used later on to get its UID.